evalml/tests/model_understanding_tests/test_partial_dependence.py

```diff
@@ -2934,3 +2935,67 @@ def test_partial_dependence_dfs_transformer_target_in_features(fast_mode, X_y_bi
 
     assert part_dep.feature_values.notnull().all()
     assert part_dep.partial_dependence.notnull().all()
+
+
+@patch(
+    "evalml.pipelines.components.transformers.preprocessing.featuretools.calculate_feature_matrix",
+)
+def test_partial_dependence_dfs_transformer_does_not_calculate_feature_matrix(
+    mock_calculate_feature_matrix,
+    X_y_binary,
+):
+    """Tests that the DFS Transformer doesn't ever have to call calculate feature matrix
+    in partial dependence fast mode. This is important, because it ensures that we are doing
+    the exact same calculations as slow mode."""
+    X, y = X_y_binary
+    X = pd.DataFrame(X)
+    X.columns = X.columns.astype(str)
+
+    es = ft.EntitySet()
+    es = es.add_dataframe(
+        dataframe_name="data",
+        dataframe=X,
+        index="index",
+        make_index=True,
+    )
+    X_fm, features = ft.dfs(
+        entityset=es,
+        target_dataframe_name="data",
+        trans_primitives=["absolute", "add_numeric"],
+    )
+
+    automl = AutoMLSearch(
+        X_train=X_fm,
+        y_train=y,
+        problem_type="binary",
+        optimize_thresholds=False,
+        max_iterations=2,
+        features=features,
+        automl_algorithm="default",
+    )
+
+    automl.search()
+
+    assert not mock_calculate_feature_matrix.called
+    pipeline = automl.get_pipeline(1)
+    pipeline.fit(X_fm, y)
+    part_dep = partial_dependence(
+        pipeline,
+        X_fm,
+        features=0,
+        grid_resolution=2,
+    )
+    fast_part_dep = partial_dependence(
+        pipeline,
+        X_fm,
+        features=0,
+        grid_resolution=2,
+        fast_mode=True,
+        X_train=X_fm,
+        y_train=y,
+    )
+    assert not mock_calculate_feature_matrix.called
+
+    assert part_dep.feature_values.notnull().all()
+    assert part_dep.partial_dependence.notnull().all()
+    pd.testing.assert_frame_equal(part_dep, fast_part_dep)
```

Comment:
Another question just for my own understanding: does this test fail with our current main branch?

Reply:
Yes, it does fail, as `calculate_feature_matrix` gets called in fast mode causing the mock checks above to fail

Reply:
Ah gotcha. I guess I'm more concerned about the `pd.testing.assert_frame_equal(part_dep, fast_part_dep)` line specifically, since that's what this PR is aimed at combatting